python/tskit/tables.py

```diff
@@ -582,6 +582,20 @@ def append(self, row):
             }
         )
 
+    def replace_with(self, other):
+        """
+        Entirely replace the contents of this table with another
+        """
+        if type(self) != type(other):
+            raise TypeError("replacement table is not of the same type")
+        params = {column: getattr(other, column) for column in self.column_names}
+        try:
```

Comment:
If you go with my last comment then the logic is inverted here - we should be checking if `self` needs `metadata_schema` not if `other` has it.